R/semver-handling.R

```diff
@@ -0,0 +1,93 @@
+#' Get max version
+#' 
+#' Extract max version from vector of version strings.
+#' 
+#' @param version_strings vector of string containing version string.
+#' 
+#' @return string representing the max version found
+get_max_version <- function(versions_strings) {
+  versions_list <- list()
+  for (version_index in seq_len(length(versions_strings))) {
+    versions_list[[version_index]] <-
+      convert_to_semver(versions_strings[[version_index]])
+  }
+  # R does not appear to contain a do while and this is the suggested approach
+  version_element_index <- 2
+  max_version <- list()
+  repeat {
```

Comment:
I'm confused how this looping is working, can you make a flow diagram and send it to me